### PR TITLE
Draft: Add ray tracing best practice guidance

### PR DESCRIPTION
**Pull request by r-potter**
@dgkoch For discussion in RT TSG

Addresses internal issues #2046, #2102, #2674

### CODE REVIEW

**Comment by gpx1000** [comment]
If you grab the latest, the CI build should succeed now.

File: chapters/extensions/ray_tracing.adoc [hunk 1]
```diff
@@ -226,3 +226,39 @@ the operation:
...
+
+All three of these structures consume driver-managed memory, the total quantity of
+which may scale based on the size of the structures themselves, the number
+of concurrently active rays, and additional factors such as levels of recursion.   
+
+Shaders should aim to keep the size of these structures low. 
+
+=== Prefer Device-Local Memory
+
+Whilst acceleration structures can be built on any Vulkan memory heap,
```
Comment (dgkoch) [inline chapters/extensions/ray_tracing.adoc:258]:
```suggestion
While acceleration structures can be built on any Vulkan memory heap,
```
(Prefer not to use Whilst)

**Review by dgkoch — DISMISSED** [review]
TSG is fine with the proposed best practices, other than the one suggestion to Americanize it :-)

**Review by dgkoch — DISMISSED** [review]
LGTM

**Comment by r-potter** [comment]
Noting that this was approved by RT TSG/Daniel. Approvals just got nuked by my pushing Steve's CI fixes

**Review by gpx1000 — APPROVED** [review]

File: chapters/extensions/ray_tracing.adoc [hunk 2]
```diff
@@ -226,3 +226,39 @@ the operation:
...
+possible to use a single ray query object even if tracing multiple rays since a
+terminated ray that's issuing a new ray can use the same object. Multiple ray
+queries in the same shader should only be needed when multiple traversals need
+to be active concurrently, and shaders should be designed to minimize the number
+of active traversals.
+
+=== Minimize the Size of Ray Payloads, Hit Attributes and Callable Data
+
+The ray tracing shader stages can communicate parameters and results using ray
+payload structures between all traversal stages, ray attribute structures
```
Comment (spencer-lunarg) [inline chapters/extensions/ray_tracing.adoc:246]: as pointed out in https://github.com/KhronosGroup/Vulkan-Guide/pull/209 

should `ray attribute structures` be `hit attribute structures`? @dgkoch @r-potter

Comment (r-potter) [reply chapters/extensions/ray_tracing.adoc:246]: @spencer-lunarg Yes it should. That's what I get for shamelessly stealing a sentence from the Khronos RT announcement blog post 😀

**Pull request merged**